compiler/rustc_typeck/src/astconv/mod.rs

```diff
@@ -522,15 +522,17 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
         &self,
         trait_ref: &hir::TraitRef<'_>,
         self_ty: Ty<'tcx>,
-    ) -> ty::TraitRef<'tcx> {
+    ) -> Option<ty::TraitRef<'tcx>> {
         self.prohibit_generics(trait_ref.path.segments.split_last().unwrap().1);
 
-        self.ast_path_to_mono_trait_ref(
-            trait_ref.path.span,
-            trait_ref.trait_def_id().unwrap_or_else(|| FatalError.raise()),
-            self_ty,
-            trait_ref.path.segments.last().unwrap(),
-        )
+        trait_ref.def_id.map(|def_id| {
```

Comment:
I'm changing semantics here a bit - if a TraitRef contains None for def_id, it'll be ignored, instead of raising a fatal error. As far as I can see, if the code contains such a case, a compile error will be emitter anyway.